Lateral Movement/ServiceAccountsPerformingRemotePS.txt

```diff
@@ -0,0 +1,55 @@
+    'Set-StrictMode',
+    'TabExpansion2'
+);
+let WhitelistedAccounts = pack_array('FakeWhitelistedAccount');     // List of accounts that are known to perform this activity in the environment and can be ignored
+LogonEvents                                                         // Get all logon events...
+| where AccountName !in (WhitelistedAccounts)                       // ...where it is not a whitelisted account...
+| where ActionType == "LogonSuccess"                                // ...and the logon was successful...
+| where AccountName !contains "$"                                   // ...and not a machine logon.
+| where AccountName !has "winrm va_"                                // WinRM will have pseudo account names that match this if there is an explicit permission for an admin to run the cmdlet, so assume it is good.
+| extend IsInteractive=(LogonType in (InteractiveTypes))            // Determine if the logon is interactive (True=1,False=0)...
+| summarize HasInteractiveLogon=max(IsInteractive)                  // ...then bucket and get the maximum interactive value (0 or 1)...
+            by AccountName, AccountDomain                           // ... by the AccountNames and DomainNames
+| where HasInteractiveLogon == 0                                    // ...and filter out all accounts that had an interactive logon.
+// At this point, we have a list of accounts that we believe to be service accounts
+// Now we need to find RemotePS sessions that were spawned by those accounts
+// Note that we look at all powershell cmdlets executed to form a 29-day baseline to evaluate the data on today
+| join kind=rightsemi (                                             // Start by dropping the account name and only tracking the...
+	MiscEvents                                                      // ...
+	| where ActionType == 'PowerShellCommand'                       // ...PowerShell commands seen...
+	| where InitiatingProcessFileName =~ 'wsmprovhost.exe'          // ...whose parent was wsmprovhost.exe (RemotePS Server)...
+    | extend AccountName = InitiatingProcessAccountName             // ...and add an AccountName field so the join is easier...
+    | extend AccountDomain = InitiatingProcessAccountDomain         // ...and add an AccountDomain field so the join is easier
+) on AccountName, AccountDomain
+// At this point, we have all of the commands that were ran by service accounts
+| extend Command = tostring(extractjson('$.Command', AdditionalFields))   // Extract the actual PowerShell command that was executed
+| where Command !in (WhitelistedCmdlets)                            // Remove any values that match the whitelisted cmdlets
+| summarize EventTime=max(EventTime), min(EventTime),               // Then group all of the cmdlets and calculate the min/max times of execution...
+    makeset(Command), count(), ReportId=max(ReportId) by            // ...as well as creating a list of cmdlets ran and the count..
```

Comment:
Why do you calculate max(ReportId)? How is it helpful?

Reply:
ReportId is a required field for custom detections.  This query can be dropped straight into a custom detection and run as-is (now that the typo in the whitelisted account is fixed).